homeassistant/components/yale_smart_alarm/alarm_control_panel.py

```diff
@@ -46,66 +61,124 @@
+    """Import Yale configuration from YAML."""
+    LOGGER.warning(
+        "Loading Yale Alarm via platform setup is depreciated; Please remove it from your configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
 
-    try:
-        client = YaleSmartAlarmClient(username, password, area_id)
-    except AuthenticationError:
-        _LOGGER.error("Authentication failed. Check credentials")
-        return
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up the alarm entry."""
 
-    add_entities([YaleAlarmDevice(name, client)], True)
+    async_add_entities(
+        [YaleAlarmDevice(coordinator=hass.data[DOMAIN][entry.entry_id]["coordinator"])]
+    )
 
 
-class YaleAlarmDevice(AlarmControlPanelEntity):
+class YaleAlarmDevice(CoordinatorEntity, AlarmControlPanelEntity):
     """Represent a Yale Smart Alarm."""
 
-    def __init__(self, name, client):
-        """Initialize the Yale Alarm Device."""
-        self._name = name
-        self._client = client
-        self._state = None
+    coordinator: YaleDataUpdateCoordinator
 
-        self._state_map = {
-            YALE_STATE_DISARM: STATE_ALARM_DISARMED,
-            YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
-            YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
-        }
+    _state_map = {
+        YALE_STATE_DISARM: STATE_ALARM_DISARMED,
+        YALE_STATE_ARM_PARTIAL: STATE_ALARM_ARMED_HOME,
+        YALE_STATE_ARM_FULL: STATE_ALARM_ARMED_AWAY,
+    }
+
+    _state = STATE_UNAVAILABLE
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self.coordinator.entry.data[CONF_NAME]
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return str(self.coordinator.entry.entry_id)
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device information about this entity."""
+        return {
+            ATTR_NAME: self.coordinator.entry.data[CONF_NAME],
+            ATTR_MANUFACTURER: MANUFACTURER,
+            ATTR_MODEL: MODEL,
+            ATTR_IDENTIFIERS: {(DOMAIN, self.coordinator.entry.data[CONF_USERNAME])},
+        }
 
     @property
     def state(self):
         """Return the state of the device."""
         return self._state
 
+    @property
+    def code_arm_required(self):
+        """Whether the code is required for arm actions."""
+        return False
+
     @property
     def supported_features(self) -> int:
         """Return the list of supported features."""
         return SUPPORT_ALARM_ARM_HOME | SUPPORT_ALARM_ARM_AWAY
 
-    def update(self):
-        """Return the state of the device."""
-        armed_status = self._client.get_armed_status()
+    async def _async_set_arm_state(self, state) -> None:
```

Comment:
I don't see the point in making these async if you are running sync functions underneath. You will not get any benefit, you may as well make these sync. If, for some reason you need it to be async, you should wrap all the synchronous calls in a single function and then create a single executor job in this async function

Reply:
Point taken. Has been changed